rasa/nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -220,51 +315,100 @@ def train(
             # create spacy lemma_ for OOV_words
             self.OOV_words = [t.lemma_ for w in self.OOV_words for t in spacy_nlp(w)]
 
-        self.vectorizer = CountVectorizer(
-            token_pattern=self.token_pattern,
-            strip_accents=self.strip_accents,
-            lowercase=self.lowercase,
-            stop_words=self.stop_words,
-            ngram_range=(self.min_ngram, self.max_ngram),
-            max_df=self.max_df,
-            min_df=self.min_df,
-            max_features=self.max_features,
-            analyzer=self.analyzer,
+        self.vectorizer = self.create_vectorizers(
+            self.use_shared_vocab,
+            self.token_pattern,
+            self.strip_accents,
+            self.lowercase,
+            self.stop_words,
+            (self.min_ngram, self.max_ngram),
+            self.max_df,
+            self.min_df,
+            self.max_features,
+            self.analyzer,
         )
 
-        lem_exs = [
-            self._get_message_text(example) for example in training_data.intent_examples
-        ]
-
-        self._check_OOV_present(lem_exs)
+        cleaned_attribute_texts = {}
+
+        for attribute in MESSAGE_ATTRIBUTES:
+            attribute_texts = [
+                self._get_message_text_by_attribute(example, attribute)
+                if example.get(attribute) is not None
+                else ""
+                for example in training_data.intent_examples
+            ]
+            self._check_OOV_present(attribute_texts)
+            cleaned_attribute_texts[attribute] = attribute_texts
+
+        combined_cleaned_texts = []
+        if self.use_shared_vocab:
```

Comment:
please separate the fitting of the vectorizers into different functions (one for shared one for not shared). this code is hard to follow